azure-pipelines/end-to-end-tests-dir/registries.ps1

```diff
@@ -201,7 +201,7 @@ try
 
     Run-Vcpkg install @builtinRegistryArgs '--feature-flags=registries,manifests'
     Throw-IfFailed
-    Require-FileEquals $installRoot/vcpkg/vcpkg-lock.json "{`n  $(ConvertTo-Json $gitRegistryUpstream): `"$gitBaselineCommit`"`n}`n"
+    Require-FileEquals $installRoot/vcpkg/vcpkg-lock.json "{`n  $(ConvertTo-Json "$gitRegistryUpstream@HEAD"): `"$gitBaselineCommit`"`n}`n"
```

Comment:
This is what I would suggest changing the file format to:

```suggestion
    Require-FileEquals $installRoot/vcpkg/vcpkg-lock.json "{`n  $(ConvertTo-Json "$gitRegistryUpstream"): {`n    `"HEAD`": `"$gitBaselineCommit`"`n  }`n}`n"
```

Reply:
Done.